cache_test.go

```diff
@@ -280,6 +268,62 @@ func testCache(t *testing.T, when spec.G, it spec.S) {
 				assertCacheHasLayer(t, testCache, "buildpack.id:layer-2")
 			})
 		})
+
+		when("buildpack API < 0.6", func() {
+			it.Before(func() {
+				exporter.Buildpacks = []buildpack.GroupBuildpack{{ID: "old.buildpack.id", API: "0.5"}}
+			})
+			when("the layers are valid", func() {
+				it.Before(func() {
+					layerFactory.EXPECT().
+						DirLayer(gomock.Any(), gomock.Any()).
+						DoAndReturn(func(id string, dir string) (layers.Layer, error) {
+							return createTestLayer(id, tmpDir)
+						}).AnyTimes()
+
+					layersDir = filepath.Join("testdata", "cacher", "layers")
+				})
+
+				when("there are previously cached layers", func() {
+					when("the SHAs match", func() {
+						it.Before(func() {
+							metadataTemplate := `{
+					"buildpacks": [
+					 {
+					   "key": "old.buildpack.id",
+					   "layers": {
+					     "cache-true-layer": {
+					       "cache": true,
+					       "sha": "%s",
+					       "data": {"old":"data"}
+					     }
+					   }
+					 }
+					]
+					}`
```

Comment:
I think we should either remove the deep indentation here (so it's realistic) or indent it further (so it looks nice).

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/537/commits/10e7898473c8b663ac444985d8b6fe429f57b8ba.